### ISSUE TITLE
prefer-const autofix causes syntax error under certain circumstances

### ISSUE THREAD

**Issue by Alaneor** [eslint/eslint#7712]
**Tell us about your environment**

* **ESLint Version:** 3.11.1
* **Node Version:** 7.2.0
* **npm Version:** 4.0.2

**What parser (default, Babel-ESLint, etc.) are you using?** `babel-eslint`

**Please show your full configuration:**

```
module.exports = {
  rules: {
    'prefer-const': 1,
  }
}
```

**What did you do? Please include the actual source code causing the issue.**

```js
function fn() {
  let body
  const body = {}
}
```

**What did you expect to happen?**

I expected to see a syntax error:

`SyntaxError: Identifier 'body' has already been declared`

This is the actual behaviour when I try to `let body` and `const body = {}` in Node.js repl. However, it's possible that this can only be determined at runtime. In that case, I would have expected the autofix to do nothing, or at least not cause this weird fix to appear.

**What actually happened? Please include the actual, raw output from ESLint.**

The autofixer attempted to "fix" the code, with the following output (which leads to syntax error):

```js
function fn() {
  let body
  constst body = {}
}
```

**Comment by not-an-aardvark** [comment]
Thanks for the report. I was able to reproduce this issue.

**Comment by not-an-aardvark** [comment]
It looks like this is caused by a bug in Acorn: https://github.com/ternjs/acorn/issues/487

The rule assumes that variables declared with `let` can only be declared in one place. However, the parser incorrectly considers the code to be valid, so the rule's assumption is violated and it behaves incorrectly.

For now, I suppose we can put a sanity check in `prefer-const` to avoid reporting an error if a variable has more than one declaration.

**Comment by platinumazure** [comment]
@not-an-aardvark At least, it should be easy to read the token we're trying to replace to make sure it actually says `let`, no?

**Comment by not-an-aardvark** [comment]
True, but there are also parts of the logic for *detecting* errors that expect only one declaration per variable. I think it's better to add an assertion in that logic to avoid potential runtime errors, rather than just when producing a fix.
